pkg/kubectl/cmd/util/factory.go

```diff
@@ -308,9 +308,16 @@ type schemaClient interface {
 	Get() *client.Request
 }
 
+func substituteUserHome(dir string) string {
+	if len(dir) == 0 || dir[0] != '~' {
```

Comment:
probably want to do a path split and check of the first segment to avoid replacing in strings like `~mydir`